springboot/src/main/java/ca/jrvs/apps/trading/dao/JdbcCrudDao.java

```diff
@@ -0,0 +1,128 @@
+package ca.jrvs.apps.trading.dao;
+
+import ca.jrvs.apps.trading.model.domain.Entity;
+import ca.jrvs.apps.trading.model.domain.Quote;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataRetrievalFailureException;
+import org.springframework.dao.IncorrectResultSizeDataAccessException;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.BeanPropertySqlParameterSource;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public abstract class JdbcCrudDao<T extends Entity<Integer>> implements CrudRepository<T, Integer> {
+
+    private static final Logger logger = LoggerFactory.getLogger(JdbcCrudDao.class);
+
+    abstract public JdbcTemplate getJdbcTemplate();
+
+    abstract public SimpleJdbcInsert getSimpleJdbcInsert();
+
+    abstract public String getTableName();
+
+    abstract public String getIdColumnName();
+
+    abstract Class<T> getEntityClass();
+
+    /**
+     * Save an entity and update auto-generated integer ID
+     * @param entity to be saved
+     * @return saved entity
+     */
+    @Override
+    public <S extends T> S save(S entity) {
+        if (existsById(entity.getId())) {
+            if (updateOne(entity) != 1) {
+                throw new DataRetrievalFailureException("Unable to retrieve entity");
+            }
+        }
+        else {
+            addOne(entity);
+        }
+        return entity;
+    }
+
+    /**
+     * helper method that saves one quote
+     */
```

Comment:
please use proper javadoc including input/output params . use IDE to generate JavaDoc template for you